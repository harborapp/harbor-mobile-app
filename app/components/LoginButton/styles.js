import { StyleSheet } from 'react-native'

import * as colors from '../../constants/colors'

export default StyleSheet.create({
  loginContainer: {
    flex: 1,
    width: '100%',
    padding: 15,
    borderRadius: 5,
    marginVertical: 8,
    backgroundColor: colors.WHITE,
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'center',
    elevation: 1
  }
})
