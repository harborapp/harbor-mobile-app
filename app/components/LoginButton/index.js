import React from 'react'
import PropTypes from 'prop-types'
import { TouchableOpacity } from 'react-native'
import Icon from 'react-native-vector-icons/FontAwesome'

import styles from './styles'
import Container from '../Container'

/**
 * LoginButton is a component to show the 'Login with <provider>' button.
 * Expects the text to be passed as a child element. It does not deal
 * with showing of the text itself, only the logo and associated styles.
 *
 * @returns {jsx}
 */
const LoginButton = ({ onPress, name, children, iconStyle }) => {
  return (
    <TouchableOpacity style={styles.loginContainer}>
      <Container>
        <Icon name={name} size={35} style={iconStyle} />
      </Container>
      <Container flex={3} style={styles.text}>
        {children}
      </Container>
    </TouchableOpacity>
  )
}

LoginButton.propTypes = {
  iconStyle: PropTypes.any,
  children: PropTypes.element.isRequired,
  onPress: PropTypes.func.isRequired,
  name: PropTypes.oneOf(['google', 'github']).isRequired
}

LoginButton.defaultProps = {
  iconStyle: {}
}

export default LoginButton
