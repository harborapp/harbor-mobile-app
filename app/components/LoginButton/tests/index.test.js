/* eslint-env jest */
import React from 'react'
import { Text } from 'react-native'
import renderer from 'react-test-renderer'

import LoginButton from '../'

describe('LoginButton', () => {
  it('should render google login button', () => {
    const tree = renderer.create(
      <LoginButton
        onPress={() => true}
        name='google'
      >
        <Text>Login with Google</Text>
      </LoginButton>
    )

    expect(tree).toMatchSnapshot()
  })

  it('should render github login button', () => {
    const tree = renderer.create(
      <LoginButton
        onPress={() => true}
        name='github'
      >
        <Text>Login with Github</Text>
      </LoginButton>
    )

    expect(tree).toMatchSnapshot()
  })

  it('should render login button with custom icon style', () => {
    const tree = renderer.create(
      <LoginButton
        onPress={() => true}
        name='github'
        iconStyle={{ color: 'red' }}
      >
        <Text>Login with Github</Text>
      </LoginButton>
    )

    expect(tree).toMatchSnapshot()
  })
})
