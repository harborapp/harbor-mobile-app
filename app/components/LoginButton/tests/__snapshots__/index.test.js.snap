// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginButton should render github login button 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#FFFFFF",
      "borderRadius": 5,
      "elevation": 1,
      "flex": 1,
      "flexDirection": "row",
      "justifyContent": "center",
      "marginVertical": 8,
      "opacity": 1,
      "padding": 15,
      "width": "100%",
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 35,
          },
          Object {},
          Object {
            "fontFamily": "FontAwesome",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      
    </Text>
  </View>
  <View
    style={
      Object {
        "flex": 3,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      Login with Github
    </Text>
  </View>
</View>
`;

exports[`LoginButton should render google login button 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#FFFFFF",
      "borderRadius": 5,
      "elevation": 1,
      "flex": 1,
      "flexDirection": "row",
      "justifyContent": "center",
      "marginVertical": 8,
      "opacity": 1,
      "padding": 15,
      "width": "100%",
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 35,
          },
          Object {},
          Object {
            "fontFamily": "FontAwesome",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      
    </Text>
  </View>
  <View
    style={
      Object {
        "flex": 3,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      Login with Google
    </Text>
  </View>
</View>
`;

exports[`LoginButton should render login button with custom icon style 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#FFFFFF",
      "borderRadius": 5,
      "elevation": 1,
      "flex": 1,
      "flexDirection": "row",
      "justifyContent": "center",
      "marginVertical": 8,
      "opacity": 1,
      "padding": 15,
      "width": "100%",
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 35,
          },
          Object {
            "color": "red",
          },
          Object {
            "fontFamily": "FontAwesome",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      
    </Text>
  </View>
  <View
    style={
      Object {
        "flex": 3,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      Login with Github
    </Text>
  </View>
</View>
`;
